src/file-events/cpp/apple_fsnotifier.cpp

```diff
@@ -89,9 +89,11 @@ Server::Server(JNIEnv* env, jobject watcherCallback, long latencyInMillis)
 }
 
 Server::~Server() {
-    vector<u16string> paths(watchPoints.size());
-    for (auto& watchPoint : watchPoints) {
-        paths.push_back(watchPoint.first);
+    vector<u16string> paths;
+    paths.reserve(watchPoints.size());
+    for (auto& it : watchPoints) {
+        auto& path = it.first;
```

Comment:
Is the difference to the previous code that `path` now is a `value_type&&` instead of a `valueTyep&`? So it is copied instead of moved?

Reply:
No, the difference is a couple lines higher. Please see the commit comment on 8bbd66b9c51f5bba1d4b30956fcb59bf13eb4ae1. The fix is to call `vector::reserve()` separately.

Reply:
Ok, so this documentation:
> (2) fill constructor
Constructs a container with n elements. Each element is a copy of val (if provided).

actually means that the container has `n` elements, not that it has enough space for `n` elements. 🤦‍♂